sources/packages/api/src/services/education-program-offering/education-program-offering.service.ts

```diff
@@ -673,17 +707,198 @@ export class EducationProgramOfferingService extends RecordDataModelService<Educ
       .where("application.applicationStatus != :cancelledStatus", {
         cancelledStatus: ApplicationStatus.cancelled,
       })
-      .andWhere("offering.id = :offeringId")
-      .setParameters({
-        cancelledStatus: ApplicationStatus.cancelled,
-        offeringId,
-      })
-      .groupBy("offering.id");
+      .andWhere("offering.id = :offeringId", { offeringId });
 
     const precedingOffering = await query.getRawOne();
 
     return {
       applicationsCount: +precedingOffering?.applicationsCount,
     };
   }
+
+  /**
+   * Get applications that are impacted by an offering change
+   * to submit reassessment.
+   * @param offeringId offering requested for change.
+   * @returns applications.
+   */
+  async getApplicationsToSubmitReassessment(
+    offeringId: number,
+  ): Promise<ApplicationAssessmentSummary[]> {
+    const applications = await this.dataSource
+      .getRepository(Application)
+      .createQueryBuilder("application")
+      .select("application.id")
+      .addSelect("assessment.id")
+      .addSelect("assessment.assessmentData IS NULL", "hasAssessmentData")
+      .addSelect("application.data->>'workflowName'", "workflowName")
```

Comment:
what is `application.data->>'workflowName'`, for my info

Reply:
Yes.